gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergPartitionDatasetFinder.java

```diff
@@ -38,6 +38,9 @@
 public class IcebergPartitionDatasetFinder extends IcebergDatasetFinder {
   public static final String ICEBERG_PARTITION_NAME_KEY = "partition.name";
   public static final String ICEBERG_PARTITION_VALUE_KEY = "partition.value";
+  public static final String ICEBERG_DATASET_VALIDATE_STRICT_PARTITION_EQUALITY = ICEBERG_DATASET_PREFIX + "validate.strict.partition.equality";
```

Comment:
considering again the naming through fresh eyes, it ought to be `partition.validate.strict.equality`, so the fully-qualified form begins w/ `iceberg.dataset.partition.`.  that's essentially the type of iceberg dataset here and if we later have additional configs for iceberg-partition copy, that should be the common prefix for them too.

the current alt. here mixes the `.partition.` in closer to the end, where the semantics get lost